lib/scanner/adapters/block-device.ts

```diff
@@ -158,7 +158,7 @@ export class BlockDeviceAdapter extends Adapter {
 						// Exclude drives with no size
 						typeof drive.size === 'number' &&
 						// Exclude virtual drives (DMG, TimeMachine, ... on macOS)
-						!drive.isVirtual
+						(process.env.ETCHER_ALLLOW_VIRTUAL_DRIVE || !drive.isVirtual)
```

Comment:
Referring to the environment so deep into a codebase is not good practice. Code must explicitly provide an API for functionality which is then configured in a “top down” fashion — consulting the environment should end at the top most layer.

Two lines above, the “includeSystemDrives()” call, gets it right and is a good way to do it.

Reply:
fair point